### ISSUE TITLE
NPE in TypeParameterUnusedInFormals

### ISSUE THREAD

**Issue by JakeWharton** [google/error-prone#282]
```
error: An unhandled exception was thrown by the Error Prone static analysis plugin.
  Please report this at https://code.google.com/p/error-prone/issues/entry and include the following:

  error-prone version: 2.0-SNAPSHOT
  Stack Trace:
  java.lang.NullPointerException
  	at com.google.errorprone.bugpatterns.TypeParameterUnusedInFormals$CastFinder.find(TypeParameterUnusedInFormals.java:276)
  	at com.google.errorprone.bugpatterns.TypeParameterUnusedInFormals.matchMethod(TypeParameterUnusedInFormals.java:116)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:789)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:138)
  	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:800)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:117)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:43)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:81)
  	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:91)
  	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:99)
  	at com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:133)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:560)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:138)
  	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:720)
  	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:50)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:78)
  	at com.google.errorprone.ErrorProneAnalyzer.reportReadyForAnalysis(ErrorProneAnalyzer.java:153)
  	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:124)
  	at com.google.errorprone.ErrorProneJavacJavaCompiler.postFlow(ErrorProneJavacJavaCompiler.java:120)
  	at com.google.errorprone.ErrorProneJavacJavaCompiler.flow(ErrorProneJavacJavaCompiler.java:92)
  	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  	at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  	at com.sun.tools.javac.main.Main.compile(Main.java:523)
  	at com.sun.tools.javac.main.Main.compile(Main.java:381)
  	at com.sun.tools.javac.main.Main.compile(Main.java:370)
  	at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:210)
  	at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:148)
```
Not sure what other information is of use to help diagnose the problem.

Using Java 1.8.0_25 and running inside Gradle 2.2.1.

**Comment by cushon** [comment]
Thanks, I think I have a fix for this.

**Comment by cushon** [comment]
Fixed at 54b05f43d14b09764f047f144f168105d9ffbde1.

**Comment by JakeWharton** [comment]
:+1: :+1: :+1: :+1: :+1: :beer:
